src/dev/flang/be/interpreter2/Intrinsics.java

```diff
@@ -0,0 +1,1534 @@
+          return JavaInterface.javaObjectToInstance(jc, resultClazz);
+        });
+    putUnsafe("fuzion.java.i16_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var s = args.get(1).i16Value();
+          var js = Short.valueOf((short) s);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(js, resultClazz);
+        });
+    putUnsafe("fuzion.java.i32_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var i = args.get(1).i32Value();
+          var ji = Integer.valueOf(i);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(ji, resultClazz);
+        });
+    putUnsafe("fuzion.java.i64_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var l = args.get(1).i64Value();
+          var jl = Long.valueOf(l);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(jl, resultClazz);
+        });
+    putUnsafe("fuzion.java.f32_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var f32 = args.get(1).f32Value();
+          var jf = Float.valueOf(f32);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(jf, resultClazz);
+        });
+    putUnsafe("fuzion.java.f64_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var d = args.get(1).f64Value();
+          var jd = Double.valueOf(d);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(jd, resultClazz);
+        });
+    putUnsafe("fuzion.java.bool_to_java_object", (interpreter, innerClazz) -> args ->
+        {
+          var b = args.get(1).boolValue();
+          var jb = Boolean.valueOf(b);
+          Clazz resultClazz = innerClazz.resultClazz();
+          return JavaInterface.javaObjectToInstance(jb, resultClazz);
+        });
+    put("fuzion.sys.internal_array_init.alloc", (interpreter, innerClazz) -> args ->
+        {
+          return ArrayData.alloc(/* size */ args.get(1).i32Value(),
+                                     /* type */ innerClazz._outer);
+        });
+    put("fuzion.sys.internal_array.get", (interpreter, innerClazz) -> args ->
+        {
+          return ((ArrayData)args.get(1)).get(
+                                   /* index */ args.get(2).i32Value(),
+                                   /* type  */ ArrayData.elementType(innerClazz._outer));
+        });
+    put("fuzion.sys.internal_array.setel", (interpreter, innerClazz) -> args ->
+        {
+          ((ArrayData)args.get(1)).set(
+                              /* index */ args.get(2).i32Value(),
+                              /* value */ args.get(3),
+                              /* type  */ ArrayData.elementType(innerClazz._outer));
+          return Value.EMPTY_VALUE;
+        });
```

Comment:
See my comment for `ArrayData.alloc`, these should pass the element clazz to `ArrayData.alloc`.